src/catalog/src/durable/impls/persist/state_update.rs

```diff
@@ -0,0 +1,620 @@
+                    })?,
+                    value.ok_or_else(|| {
+                        TryFromProtoError::missing_field("state_update_kind::GidMapping::value")
+                    })?,
+                ),
+                proto::state_update_kind::Kind::SystemPrivileges(
+                    proto::state_update_kind::SystemPrivileges { key, value },
+                ) => StateUpdateKind::SystemPrivilege(
+                    key.ok_or_else(|| {
+                        TryFromProtoError::missing_field("state_update_kind::SystemPrivileges::key")
+                    })?,
+                    value.ok_or_else(|| {
+                        TryFromProtoError::missing_field(
+                            "state_update_kind::SystemPrivileges::value",
+                        )
+                    })?,
+                ),
+                proto::state_update_kind::Kind::Timestamp(
+                    proto::state_update_kind::Timestamp { key, value },
+                ) => StateUpdateKind::Timestamp(
+                    key.ok_or_else(|| {
+                        TryFromProtoError::missing_field("state_update_kind::Timestamp::key")
+                    })?,
+                    value.ok_or_else(|| {
+                        TryFromProtoError::missing_field("state_update_kind::Timestamp::value")
+                    })?,
+                ),
+            },
+        )
+    }
+}
+
+impl Codec for StateUpdateKindTag {
+    type Schema = VecU8Schema;
+
+    fn codec_name() -> String {
+        "StateUpdateTag".to_string()
+    }
+
+    fn encode<B>(&self, buf: &mut B)
+    where
+        B: bytes::BufMut,
+    {
+        let bytes = self.0.to_le_bytes();
+        buf.put(bytes.as_slice());
+    }
+
+    fn decode<'a>(buf: &'a [u8]) -> Result<Self, String> {
+        assert_eq!(buf.len(), 8, "u64 is 8 bytes");
+        let tag = u64::from_le_bytes(buf[0..8].try_into().expect("asserted above"));
+        Ok(Self(tag))
+    }
+}
+
+impl mz_persist_types::columnar::Schema<StateUpdateKindTag> for VecU8Schema {
+    type Encoder<'a> = SimpleEncoder<'a, StateUpdateKindTag, Vec<u8>>;
+
+    type Decoder<'a> = SimpleDecoder<'a, StateUpdateKindTag, Vec<u8>>;
+
+    fn columns(&self) -> DynStructCfg {
+        SimpleSchema::<StateUpdateKindTag, Vec<u8>>::columns(&())
+    }
+
+    fn decoder<'a>(&self, cols: ColumnsRef<'a>) -> Result<Self::Decoder<'a>, String> {
+        SimpleSchema::<StateUpdateKindTag, Vec<u8>>::decoder(cols, |val, ret| {
+            *ret = StateUpdateKindTag::decode(val).expect("should be valid StateUpdateKindTag")
+        })
+    }
+
+    fn encoder<'a>(&self, cols: ColumnsMut<'a>) -> Result<Self::Encoder<'a>, String> {
+        SimpleSchema::<StateUpdateKindTag, Vec<u8>>::push_encoder(cols, |col, val| {
+            let mut buf = Vec::new();
+            StateUpdateKindTag::encode(val, &mut buf);
+            mz_persist_types::columnar::ColumnPush::<Vec<u8>>::push(col, &buf)
+        })
+    }
+}
```

Comment:
I think this wants to be a schema with a single `u64` column instead of one with a single `Vec<u8>` column. I'd call it `struct StateUpdateKindTagSchema;` and then make the second arg in all the stuff below a `u64` instead of a `Vec<u8>`

But there's not really a less-boilerplate way than Simple{Schema,Encoder,Decode}

Reply:
Thanks, just updated it to use `u64` directly.